docs/architecture/maui/authentication-and-authorization.md

```diff
@@ -1,9 +1,473 @@
+
+**Note:** Parameters can be specified on the Authorize attribute to restrict an API to specific users. For more information, see [Authorization](/aspnet/core/security/authorization/introduction) on the Microsoft Documentation Center.
+
+IdentityServer can be integrated into the authorization workflow so that the access tokens it provides control authorization. This approach is shown in the diagram below.
+
+:::image type="content" source="media/authorization-by-access-token.png" alt-text="Authorization by access token.":::
+
+The eShopOnContainers mobile app communicates with the identity microservice and requests an access token as part of the authentication process. The access token is then forwarded to the APIs exposed by the ordering and basket microservices as part of the access requests. Access tokens contain information about the client and the user. APIs then use that information to authorize access to their data. For information about how to configure IdentityServer to protect APIs, see [Configuring API resources](#configuring-api-resources).
+
+## Configuring IdentityServer to perform authorization
+
+To perform authorization with IdentityServer, its authorization middleware must be added to the web application's HTTP request pipeline. The middleware is added in the `ConfigureAuth` method in the web application's `Startup` class, which is invoked from the `Configure` method and is demonstrated in the following code example from the eShopOnContainers reference application:
+
+```csharp
+protected virtual void ConfigureAuth(IApplicationBuilder app)
+{
+    var identityUrl = Configuration.GetValue<string>("IdentityUrl");
+    app.UseIdentityServerAuthentication(new IdentityServerAuthenticationOptions
+    {
+        Authority = identityUrl.ToString(),
+        ScopeName = "basket",
+        RequireHttpsMetadata = false,
+    });
+}
+```
+
+This method ensures that the API can only be accessed with a valid access token. The middleware validates the incoming token to ensure that it's sent from a trusted issuer and validates that the token is valid to be used with the API that receives it. Therefore, browsing to the ordering or basket controller will return a `401 (unauthorized)` HTTP status code, indicating that an access token is required.
+
+> [!NOTE]
+> IdentityServer's authorization middleware must be added to the web application's HTTP request pipeline before adding MVC with `app.UseMvc()` or `app.UseMvcWithDefaultRoute()`.
+
+## Making access requests to APIs
+
+When making requests to the ordering and basket microservices, the access token, obtained from IdentityServer during the authentication process, must be included in the request, as shown in the following code example:
+
+```csharp
+var authToken = Settings.AuthAccessToken;
+Order = await _ordersService.GetOrderAsync(Convert.ToInt32(order.OrderNumber), authToken);
```

Comment:
```suggestion
Order = await _ordersService.GetOrderAsync(order.OrderNumber, authToken);
```

The `Convert.ToInt32` was a code smell to me, so upon further investigation, I realized that code doesn't exist anymore and instead the `order.OrderNumber` is already an `int.

Reply:
good catch. It was a string in the old version. I hated it and updated the code side, but missed it in the docs!